physionet-django/project/utility.py

```diff
@@ -196,15 +196,18 @@ def get_form_errors(form):
     return all_errors
 
 
-def serve_file(request, file_path):
+def serve_file(file_name, file_path):
     """
     Serve a file to download. file_path is the full file path of the
     file on the server
     """
-    if os.path.exists(file_path):
-        with open(file_path, 'rb') as f:
-            response = HttpResponse(f.read())
-            response['Content-Disposition'] = 'attachment; filename=' + os.path.basename(file_path)
-            return response
-    else:
+    try:
+        response = HttpResponse()
+        response['Content-Type'] = ''
+        response['Content-Disposition'] = 'attachment; filename=' + os.path.basename(file_name)
+        response['X-Accel-Redirect'] = '/protected/' + file_path
+        return response
+    except:
```

Comment:
@alistairewj will get upset if we let this through with a bare except. Could you add the exception type here? e.g. `except ValueError`